iothub/service/src/Common/Amqp/ClientWebSocketTransport.cs

```diff
@@ -295,25 +296,21 @@ private static void HandleWriteComplete(IAsyncResult result)
 
         private static bool WriteTaskDone(Task taskResult, TransportAsyncCallbackArgs args)
         {
-            IAsyncResult result = taskResult;
             args.BytesTransfered = 0; // reset bytes transferred
             if (taskResult.IsFaulted)
             {
                 args.Exception = taskResult.Exception;
                 return true;
             }
-            else if (taskResult.IsCompleted)
+
+            if (taskResult.IsCompleted)
             {
                 args.BytesTransfered = args.Count;
-                args.CompletedSynchronously = result.CompletedSynchronously;
-                return true;
-            }
-            else if (taskResult.IsCanceled)  // This should not happen since TaskCanceledException is handled in WriteAsyncCore.
-            {
+                args.CompletedSynchronously = ((IAsyncResult)taskResult).CompletedSynchronously;
                 return true;
             }
 
-            return false;
+            return taskResult.IsCanceled;
```

Comment:
Are these returns needed - line 310 and line 303? return taskResult.IsCanceled should be sufficient right?